src/fhirdefs/impliedExtensions.ts

```diff
@@ -0,0 +1,457 @@
+  MedicinalProductDefinition: 'MedicinalProduct',
+  PackagedProductDefinition: 'MedicinalProductPackaged',
+  RegulatedAuthorization: 'MedicinalProductAuthorization',
+  SubstanceDefinition: 'SubstanceSpecification'
+};
+
+// It's sad to ignore any child, but really, we don't care about these particular
+// children of the original element when building complex extensions.  Sorry kids!
+const IGNORED_CHILDREN = ['id', 'extension', 'modifierExtension'];
+
+export function isImpliedExtension(url: string) {
+  return IMPLIED_EXTENSION_REGEX.test(url);
+}
+
+export function materializeImpliedExtension(url: string, defs: FHIRDefinitions): any {
+  const match = url.match(IMPLIED_EXTENSION_REGEX);
+  if (match == null) {
+    logger.error(
+      `Cannot materialize implied extension (${url}) since the URL does not match the implied extension pattern ` +
+        'http://hl7.org/fhir/[version]/StructureDefinition/extension-[Path]} with version 1.0, 3.0, 4.0, or 5.0.'
+    );
+    return;
+  }
+
+  const [, version, id, type] = match;
+  const supplementalPackage = VERSION_TO_PACKAGE_MAP[version]; // guaranteed not to be null thanks to REGEX check above
+  const supplementalDefs = defs.getSupplementalFHIRDefinitions(supplementalPackage);
+  if (supplementalDefs == null) {
+    const extRelease = version === '1.0' ? 'r2' : `r${version[0]}`;
+    const fhirVersion = defs.fishForFHIR('StructureDefinition', Type.Resource)?.fhirVersion;
+    logger.error(
+      `Cannot materialize implied extension: ${url}.\n` +
+        'To fix this, add the following dependency to your sushi-config.yaml file:\n' +
+        `  hl7.fhir.extensions.${extRelease}: ${fhirVersion}`
+    );
+    return;
+  }
+
+  const sd = supplementalDefs.fishForFHIR(type);
+  if (sd == null) {
+    logger.error(
+      `Cannot materialize implied extension (${url}) since ${type} is not a valid resource or data type in ${supplementalPackage}`
+    );
+    return;
+  }
+
+  // NOTE: DSTU2 definitions don't populate element id, so fallback to path in that case
+  const ed = sd.snapshot.element.find((ed: any) => (ed.id ?? ed.path) === id);
+  if (ed == null) {
+    logger.error(
+      `Cannot materialize implied extension (${url}) since ${id} is not a valid id in ${type}`
+    );
+    return;
+  } else if (isUnsupported(ed)) {
+    logger.error(`Cannot materialize implied extension (${url}) since its type is Resource`);
+    return;
+  }
+
+  const ext = StructureDefinition.fromJSON(defs.fishForFHIR('Extension'));
+  applyMetadataToExtension(ext, sd, ed, version);
+  const rootElement = ext.findElement('Extension');
+  applyMetadataToElement(rootElement, ed);
+  applyContent(sd, ed, version, ext, rootElement, [], defs, supplementalDefs);
+
+  return ext.toJSON(true);
+}
+
+function isUnsupported(ed: any): boolean {
+  // According to documentation, elements with type "Resource" are not supported
+  return ed.type?.some((t: any) => t.code === 'Resource');
```

Comment:
Again getting theoretical here because I'm not sure if this case exists in the spec, but what if the type was something like `Patient`? Or another child of `Resource`?

Reply:
This is a direct implementation of a prohibition from the spec:
> There is no way to represent the few special elements that have the type "Resource" in extensions, so these are not defined (e.g. Bundle.entry.resource, DomainResource.contained, etc)

But I think the main issue is that `Resource` is abstract.  There is no way to create an implied extension that could work for any possible value of it.  Other abstract resources are not surfaced as element types, so we don't need to worry about them.  And concrete resources (like `Patient`) _can_ be represented in extensions just fine.  If it's a _reference_ to a concrete resource, though, then that can only be represented if the resource (or its equivalent) exists in the version of FHIR you're bringing it into (since we can't represent a _reference_ to a complex extension).